backend/elastic.js

```diff
@@ -113,6 +113,25 @@ class Elastic {
    * @param  {integer} max    The index of last document to retreive
    */
   async search(query, termId, min, max) {
+    // if we know that the query is of the format of a course code, we want to do a very targeted query against subject and classId: otherwise, do a regular query.
+    let courseCodePattern = /^(\s)*\w\w(\w?)(\w?)(\s)*(\d\d\d\d)?(\s)*$/i;
```

Comment:
Will make this regex a little more intelligent, but the idea is simply that if a subject appears first and (optionally) is followed by a classId, then we know it's a course code.

Reply:
This regex matches stuff like : `cs999998`. is that ok? 


Slight improvement - little shorter, same functionality:
```regex
/^\s*\w{2,4}\s*\d{4}?\s*$/i
```

Reply:
I don't think it'll be matching more than 4 numbers since I match against the end of the string after 4 digits (if I understand the regex correctly). This is exactly the cleanup I was looking for though, thanks!

Reply:
It matches that because the \w{2,4} matches "cs99", and then the \s* matches nothing, and then the \d{4} matches the rest. Try it in your browser:

```
/^\s*\w{2,4}\s*\d{4}?\s*$/i.test('cs999998')
```
Same thing with the regex you had:
```
 /^(\s)*\w\w(\w?)(\w?)(\s)*(\d\d\d\d)?(\s)*$/i.test('cs999998')
```
There's also some good online site for debugging regexes too. 


If you don't want to match that try something like this:
```
/^\s*[a-zA-Z]{2,4}\s*\d{4}?\s*$/i.test('cs999998')
```

Reply:
Jeez, good catch, I totally missed that. Yeah I'll definitely use the latter.